### PR TITLE
Rework tests for `look_up()`

### PR DESCRIPTION
**Pull request by etiennebacher**
Instead of using snapshots on subsets of `tetrapods`, I made a small dataframe which should cover many edge cases. Used Claude Opus 4.8 to get started with this dataframe and to help on some tests but I have reviewed / written and tested everything manually.

Part of #161 

## Checklist before requesting a review
- [x] I have read palaeoverse's [standards and structure](https://palaeoverse.palaeoverse.org/articles/structure-and-standards.html)
- [x] I have performed a self-review of my code.
- [ ] I have added function documentation.
- [ ] I have provided sufficient examples of implementation.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: R/look_up.R [hunk 1]
```diff
@@ -136,6 +136,9 @@ look_up <- function(
   if (!is.character(early_interval)) {
     stop("`early_interval` needs to be of type `character`")
   }
+  if (length(early_interval) != 1) {
+    stop("`early_interval` must be of length 1")
+  }
```
Comment (etiennebacher) [inline R/look_up.R:141]: Needed to prevent `covr` and `devtools` from reporting different snapshots, cf #235 and #238

**Review by willgearty — APPROVED** [review]

File: tests/testthat/test-look_up.R [hunk 2]
```diff
@@ -1,86 +1,275 @@
...
+      late = c("Capitanian", "Olenekian"),
+      early_stage = c("Capitanian", "Induan"),
+      late_stage = c("Capitanian", "Olenekian"),
+      interval_max_ma = c(265.1, 251.902),
+      interval_mid_ma = c(262.1, 249.551),
+      interval_min_ma = c(259.1, 247.2)
+    )
+  )
+
+  # either one or both interval colums are not found in the data
```
Comment (LewisAJones) [inline tests/testthat/test-look_up.R]:
```suggestion
  # either one or both interval columns are not found in the data
```

**Review by LewisAJones — CHANGES_REQUESTED** [review]
Thanks for putting this together @etiennebacher.

As the lead developer on this, I'm tagging @KEichenseer so he has the opportunity to comments/share any thoughts if he wants.

@KEichenseer, this function is probably going to be quite heavily refactored over the next couple of months. We'll be getting an issue together soon to describe these planned changes. It would be great to get your feedback on that issue when it's up.

**Review by LewisAJones — APPROVED** [review]
